pocket_coffea/lib/jets.py

```diff
@@ -217,9 +217,28 @@ def jet_selection(events, jet_type, params, leptons_collection=""):
         mask_lepton_cleaning = ak.prod(dR_jets_lep > cuts["dr_lepton"], axis=2) == 1
 
     if jet_type == "Jet":
-        # Selection on PUid. Only available in Run2 UL, thus we need to determine which sample we run over:
-        if events.metadata["year"] in ['2016_PreVFP', '2016_PostVFP','2017','2018']:
-            mask_jetpuid = (jets.puId >= cuts["puId"]["value"]) | (
+        # Selection on PUid. Only available in Run2 UL, thus we need to determine which sample we run over;
+        # Due to bug in JetPUid production, values are different for 2016 
+        # (see here https://twiki.cern.ch/twiki/bin/view/CMS/PileupJetIDUL#Trainings)
+        jetPuID_values = {
```

Comment:
This map should become part of the parameters instead of being hardcoded in the function. 

The location should be here https://github.com/PocketCoffea/PocketCoffea/blob/main/pocket_coffea/parameters/jet_scale_factors.yaml#L21,  adding a new subkey called "JetPuID_values". Then they can be used here :)

Reply:
Good point, I changed the Implementation accordingly.